targets/psa_switch/externs/psa_random.cpp

```diff
@@ -18,30 +18,45 @@
  */
 
 #include "psa_random.h"
-#include <bm/bm_sim/_assert.h>
+#include <bm/bm_sim/logger.h>
 
 #include <random>
 #include <thread>
+#include <iostream>
 
 namespace bm {
 
 namespace psa {
 
+void
+PSA_Random::init() {
+  valid_range = true;
+  min_val = min.get_uint64();
+  max_val = max.get_uint64();
+  range = max_val - min_val + 1;
+
+  if (range <= 0) {
```

Comment:
The only way this condition can be true for `range`, declared as type `uint64`, is if it is equal to 0.

I would recommend replacing this condition with `(min_val > max_val)`